### PR TITLE
[torch] [composable] [analytics] add analytics logging to PT-D composable APIs

### PR DESCRIPTION
**Pull request by colin2328**
Summary: as title

Test Plan: N/A

Differential Revision: D43376274

### CODE REVIEW

**Review by awgu — APPROVED** [review]
LGTM!

File: torch/distributed/_composable/checkpoint_activation.py [hunk 1]
```diff
@@ -212,6 +212,7 @@ def checkpoint(module: nn.Module, *, use_reentrant: bool = True) -> nn.Module:
         >>> model(torch.zeros(2, 10)).sum().backward()
 
     """
+    torch._C._log_api_usage_once("torch.distributed.checkpoint")
```
Comment (rohan-varma) [inline torch/distributed/_composable/checkpoint_activation.py:215]: nit: maybe checkpoint_activation so not to confuse it with state_dict checkpoint.

**Review by rohan-varma — APPROVED** [review]

File: torch/distributed/_composable/checkpoint_activation.py (hunk 1 above)
Comment (colin2328) [reply torch/distributed/_composable/checkpoint_activation.py:215]: @rohan-varma I personally like checkpoint_activation (or personally I've heard activation_checkpoint more frequently)
but I'm trying to match the API itself (which is called "checkpoint")

Should we rename the API itself? I think the log should match the API

**Review by fegin — APPROVED** [review]
Nice!

File: torch/distributed/_composable/checkpoint_activation.py (hunk 1 above)
Comment (colin2328) [reply torch/distributed/_composable/checkpoint_activation.py:215]: cc @mrshenli  WDYT about renaming the checkpoint() API to 'activation_checkpoint' or 'checkpoint_activation'?

Comment (rohan-varma) [reply torch/distributed/_composable/checkpoint_activation.py:215]: Yeah, possibly we should rename the API itself as well.

**Pull request closed without merging**